### PR TITLE
feat: adding list API

### PR DESCRIPTION
**Pull request by pgautier404**
Adds push front, push back, and fetch operations.

### CODE REVIEW

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php [hunk 1]
```diff
@@ -2,11 +2,25 @@
 namespace Momento\Cache\CacheOperationTypes;
 
 use Cache_client\_GetResponse;
+use Cache_client\_ListFetchResponse;
```
Comment (malandis) [inline src/Cache/CacheOperationTypes/CacheOperationTypes.php:5]: Is the generated code not in a separate namespace like `Momento_protos`?

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php [hunk 2]
```diff
@@ -322,3 +323,121 @@ class CacheDeleteResponseError extends CacheDeleteResponse
...
+    private array $values = [];
+
+    public function __construct(_ListFetchResponse $response)
+    {
+        parent::__construct();
+        if ($response->getFound())
+        {
+            foreach ($response->getFound()->getValues() as $value)
+            {
+                $this->values[] = $value;
```
Comment (cprice404) [inline src/Cache/CacheOperationTypes/CacheOperationTypes.php:369]: is this how you append to a list in PHP?  🤯 seems weird to be doing this in a loop

File: src/Cache/_ScsDataClient.php [hunk 3]
```diff
@@ -38,7 +52,6 @@ class _ScsDataClient
     private static int $TIMEOUT_MULTIPLIER = 1000000;
```
Comment (malandis) [inline src/Cache/_ScsDataClient.php:52]: What does this do? :thinking:

File: src/Cache/_ScsDataClient.php [hunk 4]
```diff
@@ -120,9 +132,90 @@ public function delete(string $cacheName, string $key) : CacheDeleteResponse
...
+        try {
+            validateCacheName($cacheName);
+            validateListName($listName);
+            $itemTtlSeconds = $ttlSeconds ? $ttlSeconds : $this->defaultTtlSeconds;
+            validateTtl($itemTtlSeconds);
+            $listPushFrontRequest = new _ListPushFrontRequest();
+            $listPushFrontRequest->setListName($listName);
+            $listPushFrontRequest->setValue($value);
+            $listPushFrontRequest->setRefreshTtl($refreshTtl);
+            $listPushFrontRequest->setTtlMilliseconds($itemTtlSeconds * 1000);
```
Comment (malandis) [inline src/Cache/_ScsDataClient.php:176]: I recommend pulling the seconds -> ms conversion to a separate function

File: tests/Cache/CacheClientTest.php [hunk 5]
```diff
@@ -372,4 +372,72 @@ public function testDelete() {
...
+        $this->assertNotEmpty($values);
+        $this->assertCount(1, $values);
+        $this->assertContains($value, $values);
+        $response = $this->client->listPushFront($this->TEST_CACHE_NAME, $listName, 'value', true, 6000);
+        $this->assertNotNull($response->asSuccess());
+        $response = $this->client->listFetch($this->TEST_CACHE_NAME, $listName);
+        $this->assertNotNull($response->asHit());
+        $values = $response->asHit()->values();
+        $this->assertNotEmpty($values);
+        $this->assertCount(2, $values);
```
Comment (cprice404) [inline tests/Cache/CacheClientTest.php:394]: nit - worth adding one more assertion to ensure the expected values in the expected order?

**Review by cprice404 — COMMENTED** [review]
looking good to me, just a couple minor questions

**Review by malandis — COMMENTED** [review]
Just a couple questions. Looks good

File: src/Cache/_ScsDataClient.php (hunk 3 above)
Comment (pgautier404) [reply src/Cache/_ScsDataClient.php:52]: There's a comment right above this explaining it.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php (hunk 1 above)
Comment (pgautier404) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php:5]: No, it generates its own namespaces which are separate from ours. See `composer.json` for more details.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php (hunk 2 above)
Comment (pgautier404) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php:369]: Yes, this is the syntax to append to a list. `getValues()` returns protobuf repeated fields that (as far as I know) I have to iterate over like this to extract the actual string values out of. This is not equivalent to `$this->values = $response->getFound()->getValues();` which returns a `TypeError`.

File: tests/Cache/CacheClientTest.php (hunk 5 above)
Comment (pgautier404) [reply tests/Cache/CacheClientTest.php]: Sure, added in 63c78c5.

File: src/Cache/CacheOperationTypes/CacheOperationTypes.php (hunk 2 above)
Comment (cprice404) [reply src/Cache/CacheOperationTypes/CacheOperationTypes.php:369]: yeah the iteration part makes sense to me if this is how you append to a list in php.  that is just, kind of the craziest syntax i've ever seen for appending to a list.  🤯

File: tests/Cache/CacheClientTest.php (hunk 5 above)
Comment (cprice404) [reply tests/Cache/CacheClientTest.php]: fantastic, thanks

**Review by cprice404 — DISMISSED** [review]
yes

File: src/Cache/_ScsDataClient.php (hunk 4 above)
Comment (pgautier404) [reply src/Cache/_ScsDataClient.php]: Good idea, that let me DRY out some other stuff too. Thanks for the suggestion. Fixed in 18c813c.

**Review by cprice404 — APPROVED** [review]

**Pull request merged**